### PR TITLE
Add ProxyManager ViewModel

### PR DESCRIPTION
**Pull request by fmontesino**
### Why?

The design is not ready yet, so at this point I just wanted to create the ViewModel logic.
We want to enable or disable the proxy according to the user's interaction.
We want some kind of validation when the user inputs the proxy data (valid IP + port numeric & in range).

### What?

Created a ProxyValidator class that uses the AndroidX pattern validation for checking if the IP and port are valid.
Added a ViewModel that exposes validation events (for now, at least) and the proxy state, mapping the liveData from the DeviceSettingsManager.

### Bonus

Refactored the Proxy class to use two Strings: it makes everything more readable, and now that we have validation makes more sense.
Added SingleLiveEvent for one-time only events.

### CODE REVIEW

File: feature-proxy-manager/src/main/java/com/kinandcarta/create/proxytoggle/manager/viewmodel/ProxyManagerViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    val proxyEvent = SingleLiveEvent<ProxyManagerEvent>()
+
+    val proxyState = Transformations.map(deviceSettingsManager.proxySetting) { proxy ->
+        if (proxy.isEnabled) {
+            ProxyState.Enabled(proxy.address, proxy.port)
+        } else ProxyState.Disabled()
+    }
+
+    fun enableProxy(address: String, port: String) {
+        if (!proxyValidator.isValidIP(address)) {
```
Comment (kleinsenberg) [inline feature-proxy-manager/src/main/java/com/kinandcarta/create/proxytoggle/manager/viewmodel/ProxyManagerViewModel.kt]: Would `when` make this more readable?

File: feature-proxy-manager/src/main/java/com/kinandcarta/create/proxytoggle/manager/model/Proxy.kt [hunk 2]
```diff
@@ -1,10 +1,10 @@
 package com.kinandcarta.create.proxytoggle.manager.model
 
-data class Proxy(val address: String, val port: Int) {
+data class Proxy(val address: String, val port: String) {
```
Comment (kleinsenberg) [inline feature-proxy-manager/src/main/java/com/kinandcarta/create/proxytoggle/manager/model/Proxy.kt:3]: Any specific reason why?

**Review by kleinsenberg — APPROVED** [review]

File: feature-proxy-manager/src/main/java/com/kinandcarta/create/proxytoggle/manager/viewmodel/ProxyManagerViewModel.kt (hunk 1 above)
Comment (fmontesino) [reply feature-proxy-manager/src/main/java/com/kinandcarta/create/proxytoggle/manager/viewmodel/ProxyManagerViewModel.kt]: I am not sure:

```kotlin
when {
    !proxyValidator.isValidIP(address) -> {
        proxyEvent.value = ProxyManagerEvent.InvalidAddress
    }
    !proxyValidator.isValidPort(port) -> {
        proxyEvent.value = ProxyManagerEvent.InvalidPort
    }
    else -> {
        deviceSettingsManager.enableProxy(Proxy(address, port))
    }
}
```

File: feature-proxy-manager/src/main/java/com/kinandcarta/create/proxytoggle/manager/model/Proxy.kt (hunk 2 above)
Comment (fmontesino) [reply feature-proxy-manager/src/main/java/com/kinandcarta/create/proxytoggle/manager/model/Proxy.kt:3]: I had no validation before, and I was converting `.toInt()` and `.toString()` all the time.
In the end I just need a String, the user will input it as a String and I added validation, so there's no point in doing conversions all the time

File: feature-proxy-manager/src/main/java/com/kinandcarta/create/proxytoggle/manager/viewmodel/ProxyManagerViewModel.kt (hunk 1 above)
Comment (fmontesino) [reply feature-proxy-manager/src/main/java/com/kinandcarta/create/proxytoggle/manager/viewmodel/ProxyManagerViewModel.kt]: Meh, I guess it's less bad. Changed

**Pull request merged**